### PR TITLE
Only show Out of Date when adding metrics not in the snapshot

### PR DESCRIPTION
**Pull request by lukesonnet**
### Features and Changes

Now only show "Out of Date" on results when adding a metric that isn't there, rather than when removing a metric (which should be fine from the perspective of the user, for the most part).

This keeps "Out of Date" a bit more meaningful in the user's eyes.

### Screenshots

Current and new behavior in one loom: https://www.loom.com/share/0bb91643f4d04b5aaf5b76ae28f07a25?sid=515255c6-e99b-4f2a-9ae4-4ae2d4abbc79

### CODE REVIEW

File: packages/front-end/components/Experiment/AnalysisSettingsBar.tsx [hunk 1]
```diff
@@ -545,6 +545,15 @@ function isDifferentStringArray(
   if (val1.length !== val2.length) return true;
   return val1.some((v) => !val2.includes(v));
 }
+function parentStringDoesNotContainAllChildrenValues(
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/Experiment/AnalysisSettingsBar.tsx:548]: should be `parentStringArray...` (array). But consider something like `isStringArrayMissingElements()` or something...

File: packages/front-end/components/Experiment/AnalysisSettingsBar.tsx [hunk 2]
```diff
@@ -545,6 +545,15 @@ function isDifferentStringArray(
...
+  parent?: string[] | null,
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/Experiment/AnalysisSettingsBar.tsx:549]: you can avoid some of the null checks by simplifying the interface:
```
parent: string[] = [],
child: string[] = [],

File: packages/front-end/components/Experiment/AnalysisSettingsBar.tsx [hunk 3]
```diff
@@ -545,6 +545,14 @@ function isDifferentStringArray(
   if (val1.length !== val2.length) return true;
   return val1.some((v) => !val2.includes(v));
 }
+function isStringArrayMissingElements(
+  strings: string[] = [],
+  elements: string[] = []
+) {
+  if (!elements) return false;
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/Experiment/AnalysisSettingsBar.tsx:552]: not needed as elements is always an array

Comment (lukesonnet) [reply packages/front-end/components/Experiment/AnalysisSettingsBar.tsx:552]: I disagree? if elements is an empty array we can return right away.

**Review by bryce-fitzsimons — APPROVED** [review]

**Pull request merged**